### ISSUE TITLE
Strict flag gives less errors than default mode

### ISSUE THREAD

**Issue by ljden** [python/mypy#12497]
**Bug Report**

When I run mypy with the `--strict` flag I do not get an error that shows up when running without any flags.

**To Reproduce**

Minimal working example:
```python
# temp.py
import flask

app = flask.Flask("name")

@app.route("/")
def home() -> flask.Response:
    return f""
```

```bash
$ mypy temp.py 
temp.py:8: error: Incompatible return value type (got "str", expected "Response")
Found 1 error in 1 file (checked 1 source file)
$ mypy --strict temp.py 
Success: no issues found in 1 source file
```
**Expected Behavior**

Expected `--strict` flag to have the same or more errors than default

**Actual Behavior**

mypy with `--strict` flag reported no errors

**Your Environment**

- Mypy version used: 0.931
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: Python 3.6.9
- Operating system and version: Linux Mint 19.3 Cinnamon

**Comment by erictraut** [comment]
I'm not able to repro this issue with the latest version of mypy (1.5). Perhaps it was fixed somewhere along the way.

**Comment by hauntsaninja** [comment]
Yeah, I think there was an issue related to `--no-implicit-export`, where mypy would treat implicitly exported symbols as not existing if you tried to import them, so you got Any
